### PR TITLE
Add check if user is allowed to assign a user to task

### PR DESCRIPTION
**Pull request by maarten-ritense**
When opening the task modal now also a check is performed if the user is allowed to assign a user to the task

### CODE REVIEW

File: projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts [hunk 1]
```diff
@@ -129,11 +145,33 @@ export class TaskDetailModalComponent implements OnDestroy {
     this.intermediateSaveEnabled = this.configService.featureToggles.enableIntermediateSave;
 
     this.iconService.registerAll([RecentlyViewed16]);
+
+    this.canAssignUserToTask$.subscribe((canAssign) => {
```
Comment (mbritense) [inline projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts]: This subscription is never closed

File: projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts [hunk 2]
```diff
@@ -101,11 +104,22 @@ export class TaskDetailModalComponent implements OnDestroy {
   public readonly processLinkIsFormFlow$ = this.taskProcessLinkType$.pipe(
     map(type => type === 'form-flow')
   );
+  public readonly canAssignUserToTask$ = new BehaviorSubject<boolean>(false)
 
   private readonly processLinkId$ = new BehaviorSubject<string>(undefined);
 
+  public readonly canAssign$: Observable<boolean> = this.task$.pipe(
+    switchMap(task =>
```
Comment (teodor-ritense) [inline projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts]:
```suggestion
    switchMap((task: Task | null) =>
```

File: projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts [hunk 3]
```diff
@@ -129,11 +145,33 @@ export class TaskDetailModalComponent implements OnDestroy {
     this.intermediateSaveEnabled = this.configService.featureToggles.enableIntermediateSave;
 
     this.iconService.registerAll([RecentlyViewed16]);
+
+    this.canAssignUserToTask$.subscribe((canAssign) => {
+      this.logger.debug("Is user allowed to assign a user to Task", canAssign);
+    })
+  }
+
+  ngOnInit(): void {
```
Comment (teodor-ritense) [inline projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts]:
```suggestion
  public ngOnInit(): void {
```

File: projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts [hunk 4]
```diff
@@ -101,11 +104,22 @@ export class TaskDetailModalComponent implements OnDestroy {
   public readonly processLinkIsFormFlow$ = this.taskProcessLinkType$.pipe(
     map(type => type === 'form-flow')
   );
+  public readonly canAssignUserToTask$ = new BehaviorSubject<boolean>(false)
 
   private readonly processLinkId$ = new BehaviorSubject<string>(undefined);
 
+  public readonly canAssign$: Observable<boolean> = this.task$.pipe(
+    switchMap(task =>
+      this.permissionService.requestPermission(CAN_ASSIGN_TASK_PERMISSION, {
```
Comment (teodor-ritense) [inline projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.ts]: Please add a safe condition for the case in which task$ returns a null

**Review by teodor-ritense — APPROVED** [review]

**Pull request merged**